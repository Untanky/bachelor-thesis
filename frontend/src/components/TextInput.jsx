import React from 'react';

import './TextInput.scss';

const TextInput = ({ label, placeholder, initialValue, onChange }) => {
  const id = Math.random();

  return (
    <div className="text-input">
      <label
        className="label"
        htmlFor={id}
      >
        {label}
      </label>
      <input
        id={id}
        className="input-field"
        type="text"
        onChange={(evt) => onChange(evt.target.value)}
        defaultValue={initialValue}
        placeholder={placeholder}
      />
    </div>
  );
};

export default TextInput;
