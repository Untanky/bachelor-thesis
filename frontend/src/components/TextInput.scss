@import "../theme";

.text-input {
  .label {
    display: block;
    text-align: left;
    padding: 0 14px;
  }

  .input-field {
    box-sizing: content-box;
    width: 100%;
    outline: none;
    padding: 0.5em 1em;
    border-radius: 2em;
    border: 1px solid #222;

    &:active, &:focus {
      border-color: $htw-green;
    }
  }
}