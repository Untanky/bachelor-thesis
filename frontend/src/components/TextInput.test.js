import React from 'react';
import TextInput from './TextInput';
import renderer from 'react-test-renderer';

global.Math.random = () => 0.5;

it('renders correctly with initial value', () => {
  const tree = renderer
    .create(
      <TextInput
        label="Test label"
        placeholder="Placeholder"
        initialValue="Initial value"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders correctly without initial value', () => {
  const tree = renderer
    .create(
      <TextInput
        label="Test label"
        placeholder="Placeholder"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
