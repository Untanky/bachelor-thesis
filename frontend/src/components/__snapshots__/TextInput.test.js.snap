// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with initial value 1`] = `
<div
  className="text-input"
>
  <label
    className="label"
    htmlFor={0.5}
  >
    Test label
  </label>
  <input
    className="input-field"
    id={0.5}
    onChange={[Function]}
    placeholder="Placeholder"
    type="text"
    value="Initial value"
  />
</div>
`;

exports[`renders correctly without initial value 1`] = `
<div
  className="text-input"
>
  <label
    className="label"
    htmlFor={0.5}
  >
    Test label
  </label>
  <input
    className="input-field"
    id={0.5}
    onChange={[Function]}
    placeholder="Placeholder"
    type="text"
    value=""
  />
</div>
`;
